filter_plugins/list-dict-diff.py

```diff
@@ -8,9 +8,62 @@ def get_remaining_items(list_a, list_b, key_a, key_b):
 
     return list_a
 
+# For manage-workflow-templates role
+# Takes a workflow defined in a dictionary and processes it into a list
+def flatten_workflow_nodes(workflows):
+    flattened_workflow, flattened_start_tree = [], []
+    current_node_index = 1
+
+    # Processes each node beginning at start and all their success/failure branches
+    for workflow in workflows:
+        flattened_start_tree = gen_flattened_list(workflow, current_node_index, 0, 'start', [])
+        current_node_index += len(flattened_start_tree)
+        flattened_workflow = flattened_workflow + flattened_start_tree
+
+    return flattened_workflow
+
+# Helper function for flatten_workflow_nodes
+# Recursively flattens a workflow dictionary with one node at start
+def gen_flattened_list(workflow, node_index, parent_index, node_type, flattened_nodes):
+    current_node = [{
+     'job_name': workflow['unified_job_template']['name'],
+     'type': node_type,
+     'index': node_index,
+     'parent_index': parent_index
+   }]
+
+   # Return node it has no children
```

Comment:
I don't think this base case is needed as you won't be processing success/failture nodes below unless there are some anyway ... just simplifies the code if this can be eliminated.